### DISCUSSION TITLE
Assistant Panel: save chat to project

### DISCUSSION THREAD

**Discussion by NiloCK** [zed-industries/zed#21486] [Feature Requests]
### Check for existing issues

- [X] Completed

### Describe the feature

This is not a fully formed idea. Maybe it's not good, but could be worth some consideration or discussion. Closely related to, but I think distinct from, https://github.com/zed-industries/zed/issues/19002.

I have often found that there's a lot of useful context in terms of reasoning about the design and growth of a system in my assistant panel histories. But after commits are made, pull requests are opened, that context is lost other than as a vague memory for the developer and as history stored locally in that developer's editor.

It **could** be useful to selectively include some chats in the project itself. Say, in a `./.zedai` or `./.llm` directory.

One could imagine a workflow where an LLM traverses this history in order to produce a digest containing all of this context, Which itself could then be fed as context for future chats.

The chat history files themselves should be annotated with, say, 
- the get commit hash at the time of the chat
- (optionally) edited so that the committing author's name/identifier is inserted for the generic `user:`
- model data

### Environment

Zed: v0.163.2 (Zed)
OS: Linux Wayland ubuntu 22.04
Memory: 15.4 GiB
Architecture: x86_64
GPU: Intel(R) UHD Graphics (CML GT2) || Intel open-source Mesa driver || Mesa 23.2.1-1ubuntu3.1~22.04.2

### If applicable, add mockups / screenshots to help present your vision of the feature

EG, a `save chat to project` option here.

![image](https://github.com/user-attachments/assets/76858e62-2344-42ce-b00e-3b0adb1b5146)

**Comment by goodlux** [comment]
This would be really useful! What I do now is have the assistant make a summary of the chat context, then park the summary in an md file that goes into git. This works me alone, wouldn't really really scale well. 

I think to some extent this could be solved by git integration into the assistant panel (through MCP server and /slash) coupled with a really good MCP server implementation that stores all the chats.

It almost seems as though some new tech that runs parallel to git is needed ... with git you have the code development, which is intended to updated, forked, changed AND then you also have the running "log" of conversation between developers themselves and AI ... and this you don't want updated, you want it preserved, but summarized and accessible.

**Reply by au-phiware** [reply]
> ...AND then you also have the running "log" of conversation between developers

I 100% thought you were talking about git commit messages here 😁

When a commit is made after some back-and-forth with the assistant, I find it useful to have the assistant summarize the context and describe the change, and write the commit message. In my mind, these commit messages should form part of the context for future chats.

Another option could be to store the chat history as a [git note](https://git-scm.com/docs/git-notes). Notes are typically used to supplement a commit with information that wasn't available at the time of the commit and can take any format. But they can be attached to a tag, tree, file, or potentially another note.

**Reply by goodlux** [reply]
oh this is an interesting feature of git ... I could see how this would be helpful for storing more context. I was thinking of something more at the repo level, but maybe conversation notes stored this way could be aggregated.

**Comment by strangemonad** [comment]
I'd love if there were a setting that could configure all agent threads to be saved as md files in the current workspace, possibly under `.zed/agent` or something like that. It would streamline cases where one thread can solve a sub-task but you need to summarize the output to use in another agent thread.

**Comment by Mayeu** [comment]
Note that with the buffer open you should be able to save it like any buffer. I do that with the normal vim save command (`:w path/to/the/file`).

But as described in the original post, I would really like to not even have to manually open the thread buffer and save it, just have all the files in a folder that I can set by project so that I can commit them with the changes that were made.

Edit: By "with the buffer open" I mean click on the small document icons close to the thumbs up/down, it open the conversation as markdown in a new buffer:
<img width="540" alt="Screenshot 2025-06-18 at 21 05 30" src="https://github.com/user-attachments/assets/19d5e938-7a08-41fa-8a6f-d277acad829b" />

**Comment by Cobertos** [comment]
It would be nice to have the timestamps too. Currently it shows a `X hours ago` when looking at old conversations, but there is no such thing to view in the markdown log itself.

**Comment by ivan-magda** [comment]
+1 on this. Right now all my AI conversations are mixed together globally, which makes it hard to find relevant context when I come back to a project after a few days.

Having threads stored per-project (or at least filtered by project in the history panel) would be a big improvement. Something like how Cursor handles it where each workspace has its own chat history.

**Comment by ninjaprox** [comment]
Currently all conversations are stored in `~/.config/zed/conversations/*.json`. We can copy the idea from Claude Code, they store conversations grouped by project paths like `~/.claude/projects/<path-to-project>`. It looks like the most straightforward approach at the moment.

**Comment by ec1oud** [comment]
https://zed.dev/docs/ai/text-threads says on Linux it should ve saving to `~/.local/share/zed/conversations/`... but why json?  I think markdown would work.  Anyway, for me they are not going there (although I do have some old json files from when it was doing that): they are saved to ~/.local/share/zed/threads/threads.db apparently as compressed blobs (it's a sqlite database).   (I have version 0.223.2-1 on Arch Linux, and have been using it with ollama)

Is it a matter of configuration to tell it where to save the chat logs?  Then I think it would be a good idea to _be able to_ save them to some directory under the project, as suggested.
